swap/src/cli/watcher.rs

```diff
@@ -40,51 +40,67 @@ impl Watcher {
     /// Start running the watcher event loop.
     /// Should be done in a new task using [`tokio::spawn`].
     pub async fn run(mut self) {
-        // Note: since this is de-facto a daemon, we have to gracefully handle errors
+        // Note: since this is a daemon, we have to gracefully handle errors
         // (which in our case means logging the error message and trying again later)
         loop {
             tokio::time::sleep(Duration::from_secs(Watcher::CHECK_INTERVAL)).await;
 
-            // Fetch current transactions and timelocks
-            let current_swaps = match self.get_current_swaps().await {
-                Ok(val) => val,
-                Err(e) => {
-                    tracing::error!(error=%e, "Failed to fetch current transactions, retrying later");
-                    continue;
-                }
-            };
+            match self.do_work().await {
+                Ok(()) => (),
+                Err(error) => tracing::error!(%error, "Watcher daemon encountered error"),
+            }
+        }
+    }
+
+    /// This function performs the actual work and get's called every [`Watcher::CHECK_INTERVAL`]
+    /// seconds.
+    ///
+    /// Splitting the content of the loop like this allows better error handling.
+    async fn do_work(&mut self) -> Result<()> {
```

Comment:
Only issue I see with this is that if one of the components of the worker fails, all other ones will as well.

E.g if we fail to get the balance for some reason, we won't be able to update our timelocks.

Reply:
I thought about that, but I don't think it's likely that only one of the components fails.